netsim/daemons/bird/bgp.j2

```diff
@@ -49,12 +49,14 @@ filter bgp_export_{{ ntype }} {
 {% endfor %}
 
 {% for n in bgp.neighbors %}
-{%   for af in [ 'ipv4','ipv6' ] if af in n and n[af] is string %}
-protocol bgp bgp_{{ n.name }}_{{ af }} {
+{%   for af in [ 'ipv4','ipv6' ] if af in n and (n[af] is string 
+                                                 or (af=='ipv6' and n.ipv4|default(False) is sameas True)) %}
+{%     set dual_unnum = n.ipv4|default(False) is sameas True and n.ipv6|default(False) is sameas True %}
+protocol bgp bgp_{{ n.name }}_{{ 'dual_unnumbered' if dual_unnum else af }} {
 {%     set loopback = loopback|default({}) %}
 {%     set local_ip = loopback[af]|default('') %}
   local {{ local_ip.split('/')[0] if n.type == 'ibgp' else '' }} as {{ n.local_as|default(bgp.as) }};
-  neighbor {{ n[af] }} as {{ n['as'] }};
+  neighbor {{ n[af] if n[af] is string else 'range ::/0' }} as {{ n['as'] }};
```

Comment:
This hack is too ugly to merge. Looks like we'll stay at RFC 8950 support with no unnumbered EBGP.

Reply:
ok, removed